<launch>

  <include file="$(find zed_wrapper)/launch/zed_tf.launch" />

  <arg name="svo_file" default=""/>
  
      <param name="robot_description" textfile="$(find zed_wrapper)/urdf/zed.urdf" />

  <group ns="camera">

    <node name="zed_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <node name="zed_wrapper_node" pkg="zed_wrapper" type="zed_wrapper_node" args="$(arg svo_file)" output="screen">

      <!-- Resolution
        0 = HD2K    2208*1242, supported framerate : 15 fps
        1 = HD1080  1920*1080, supported framerates : 15, 30 fps
        2 = HD720   1280*720, supported framerates : 15, 30, 60 fps
        3 = VGA     672*376, supported framerates : 15, 30, 60, 100 fps
      -->
      <param name="resolution"            value="2" />

      <!-- Quality
        0 = NONE          Disparity map not computed, only the rectified images will be available.
        1 = PERFORMANCE   Fastest mode, also requires less GPU memory, the disparity map is less robust
        2 = MEDIUM        Balanced quality mode, requires less GPU memory but the disparity map is a little less detailed
        3 = QUALITY       Better quality mode, the disparity map is more precise
      -->
      <param name="quality"               value="3" />

      <!-- Sensing Mode
        0 = FILL      Occlusion filling, edge sharpening, advanced post-filtering. Application example : Refocusing, Multi-view generation
        1 = STANDARD  Structure conservative, no occlusion filling. Application example : Obstacle detection, 3D reconstructions, 3D measures
      -->
      <param name="sensing_mode"          value="1" />

      <param name="frame_rate"            value="15" />
      <param name="odometry_DB"           value="" />
      <param name="openni_depth_mode"     value="1" />

      <param name="rgb_topic"             value="rgb/image_rect_color" />
      <param name="rgb_cam_info_topic"    value="rgb/camera_info" />
      <param name="rgb_frame_id"          value="/zed_tracked_frame" />

      <param name="left_topic"            value="left/image_rect_color" />
      <param name="left_cam_info_topic"   value="left/camera_info" />
      <param name="left_frame_id"         value="/zed_tracked_frame" />

      <param name="right_topic"           value="right/image_rect_color" />
      <param name="right_cam_info_topic"  value="right/camera_info" />
      <param name="right_frame_id"        value="/zed_tracked_frame" />

      <param name="depth_topic"           value="depth/image_rect_color" />
      <param name="depth_cam_info_topic"  value="depth/camera_info" />
      <param name="depth_frame_id"        value="/zed_tracked_frame" />

      <param name="point_cloud_topic"     value="point_cloud/cloud" />
      <param name="cloud_frame_id"        value="/zed_tracked_frame" />

      <param name="odometry_topic"                value="odom" />
      <param name="odometry_frame_id"             value="/zed_initial_frame" />
      <param name="odometry_transform_frame_id"   value="/zed_tracked_frame" />


    </node>
  </group>
</launch>
